### PR TITLE
Allow to build httpd24 with mod_http2

### PR DESCRIPTION
**Pull request by kabel**
The h2 module brings http/2.0 support into httpd v2.4 with a upstream
maintained patch.

### CODE REVIEW

File: mod_h2.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+class ModH2 < Formula
+  desc "Apache httpd module implementing the HTTP2 protocol."
+  homepage "https://icing.github.io/mod_h2/"
+  url "https://github.com/icing/mod_h2/releases/download/v0.9.8/mod_h2-0.9.8.tar.gz"
+  sha256 "af38805ae3c5d63df681b23b661c9d87b1e1db059553083226e961ec6cf8a873"
+  head "https://github.com/icing/mod_h2.git"
+
+  head do
+    depends_on "autoconf" => :build
```
Comment (dunn) [inline mod_h2.rb]: If there's a head block you can add the url to it:
```ruby
head do
  url "https://github.com/icing/mod_h2.git"
  depends_on "autoconf" => :build
end
```
Then you can remove `head "https://github.com/icing/mod_h2.git"` above.

**Comment by dunn** [comment]
Build failure on Yosemite and Mavericks:
```
h2_h2.c:152:20: error: use of undeclared identifier 'AP_PROTOCOL_HTTP1'
        && !strcmp(AP_PROTOCOL_HTTP1, ap_get_protocol(c))) {
                   ^
h2_h2.c:161:27: error: use of undeclared identifier 'AP_PROTOCOL_HTTP1'
                || strcmp(AP_PROTOCOL_HTTP1, ap_get_protocol(c))) {
                          ^
2 errors generated.
```

**Comment by kabel** [comment]
I don't know how to properly force/check that "--with-h2-support" is installed on httpd24. The patch at https://github.com/kabel/homebrew-apache/blob/patch-h2/httpd24.rb#L31 _must_ be applied to compile.

**Comment by dunn** [comment]
Oh, I see what you mean.  Change the `depends_on` line to: `depends_on "httpd24" => "with-h2-support"`

**Comment by kabel** [comment]
@dunn any chance this can get another look?

**Comment by dunn** [comment]
Test and audit failures on some machines: http://bot.brew.sh/job/Homebrew%20Apache%20Pull%20Requests/27/#showFailuresLink

**Comment by kabel** [comment]
Thanks. These are not failures that I introduced, but I'll see what I can do about getting rid of them.

**Comment by kabel** [comment]
I've addressed the test failures. The build continues to fail because of another outdated formula in the repo that I don't think would be appropriate to address in this PR (it is addressed in another).

**Comment by kabel** [comment]
Tests passing!

**Comment by kabel** [comment]
I'm abandoning this in favor of a cleaner history. See #103

**Pull request closed without merging**